File: SmartPhone++/SmartPhone.Services/Database/PartCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class PartCategory
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string? Description { get; set; }
        
        public string? ImageUrl { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? UpdatedAt { get; set; }
        
        // Self-referencing relationship for subcategories
        public int? ParentCategoryId { get; set; }
        public PartCategory? ParentCategory { get; set; }
        public ICollection<PartCategory> SubCategories { get; set; } = new List<PartCategory>();
        
        // Navigation property for parts
        public ICollection<Part> Parts { get; set; } = new List<Part>();
    }
} 

File: SmartPhone++/UI/smartphone_desktop_admin/lib/model/part_category.dart
import 'package:json_annotation/json_annotation.dart';

part 'part_category.g.dart';

@JsonSerializable()
class PartCategory {
  final int id;
  final String name;
  final String? description;
  final String? imageUrl;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int? parentCategoryId;
  final String? parentCategoryName;

  PartCategory({
    this.id = 0,
    this.name = '',
    this.description,
    this.imageUrl,
    this.isActive = true,
    required this.createdAt,
    this.updatedAt,
    this.parentCategoryId,
    this.parentCategoryName,
  });

  factory PartCategory.fromJson(Map<String, dynamic> json) => _$PartCategoryFromJson(json);
  Map<String, dynamic> toJson() => _$PartCategoryToJson(this);
} 